### PR TITLE
Add 'validate' function to public key type

### PR DESCRIPTION
**Pull request by SupunS**
## Description

Work towards #582

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: docs/language/accounts.mdx [hunk 1]
```diff
@@ -167,6 +167,9 @@ struct AccountKey {
 struct PublicKey {
     let publicKey: [UInt8]
     let signatureAlgorithm: SignatureAlgorithm
+
+    // Verifies the validity of this public key.
```
Comment (turbolent) [inline docs/language/accounts.mdx:171]:
```suggestion
    /// Verifies the validity of this public key
```

File: runtime/account_keys_test.go [hunk 2]
```diff
@@ -966,3 +967,95 @@ type testAccountKeyStorage struct {
 	keys        []*AccountKey
 	returnedKey *AccountKey
 }
+
+func TestPublicKey(t *testing.T) {
```
Comment (turbolent) [inline runtime/account_keys_test.go:971]: 👍

File: runtime/account_keys_test.go [hunk 3]
```diff
@@ -966,3 +967,95 @@ type testAccountKeyStorage struct {
...
+
+	t.Parallel()
+
+	rt := NewInterpreterRuntime()
+
+	runtimeInterface := &testRuntimeInterface{
+		validatePublicKey: func(publicKey *PublicKey) (bool, error) {
+			if len(publicKey.PublicKey) == 0 {
```
Comment (turbolent) [inline runtime/account_keys_test.go:979]: Maybe assert here that the function is called, by declaring a variable outside, setting it to true here, and asserting outside it was true after the execution

**Review by turbolent — APPROVED** [review]

File: runtime/account_keys_test.go (hunk 3 above)
Comment (SupunS) [reply runtime/account_keys_test.go]: That's a good idea. Added in 03fa1ce

**Pull request merged**